### ISSUE TITLE
Ref safety violation is not detected when property accessors have parameters with different EffectiveScope.

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#71798]
The following unit test succeeds today:
```
        [Fact]
        public void RefSafety()
        {
            var consumer = @"
using System;

class Program
{
    Span<long> Test()
    {
        Span<long> s = stackalloc long[3];
        return new Test()[s];
    }
}
";
            var src0 = @"
using System;

public class Test
{
    public Span<long> this [Span<long> a3]
    {
        get 
        {
            return a3;
        }
        set
        {}
    }
}
";

            var comp0 = CreateCompilation([src0, consumer], targetFramework: TargetFramework.Net80);

            var indexer0 = comp0.GetMember<PropertySymbol>("Test.this[]");

            Assert.Equal(ScopedKind.None, indexer0.Parameters[0].EffectiveScope);
            Assert.Equal(ScopedKind.None, indexer0.SetMethod.Parameters[0].EffectiveScope);
            Assert.Equal(ScopedKind.None, indexer0.GetMethod.Parameters[0].EffectiveScope);

            comp0.VerifyDiagnostics(
                // (9,16): error CS8347: Cannot use a result of 'Test.this[Span<long>]' in this context because it may expose variables referenced by parameter 'a3' outside of their declaration scope
                //         return new Test()[s];
                Diagnostic(ErrorCode.ERR_EscapeCall, "new Test()[s]").WithArguments("Test.this[System.Span<long>]", "a3").WithLocation(9, 16),
                // (9,27): error CS8352: Cannot use variable 's' in this context because it may expose referenced variables outside of their declaration scope
                //         return new Test()[s];
                Diagnostic(ErrorCode.ERR_EscapeVariable, "s").WithArguments("s").WithLocation(9, 27)
                );

            //public class Test
            //{
            //    public Span<long> this [Span<long> a3]
            //    {
            //        get 
            //        {
            //            return a3;
            //        }
            //        set [scoped Span<long> a3]
            //        {}
            //    }
            //}
            string il = @"
.assembly extern System.Runtime { .ver 8:0:0:0 .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A) }

.module '_0_6c26dab9-ae11-4e07-a2d8-3eed9fd85426.dll'
.custom instance void [System.Runtime]System.Runtime.CompilerServices.RefSafetyRulesAttribute::.ctor(int32) = ( 01 00 0B 00 00 00 00 00 ) 

.class public auto ansi beforefieldinit Test
    extends [System.Runtime]System.Object
{
    .custom instance void [System.Runtime]System.Reflection.DefaultMemberAttribute::.ctor(string) = (
        01 00 04 49 74 65 6d 00 00
    )
    // Methods
    .method public hidebysig specialname 
        instance valuetype [System.Runtime]System.Span`1<int64> get_Item (
            valuetype [System.Runtime]System.Span`1<int64> a3
        ) cil managed 
    {
        // Method begins at RVA 0x2050
        // Code size 2 (0x2)
        .maxstack 8

        IL_0000: ldarg.1
        IL_0001: ret
    } // end of method Test::get_Item

    .method public hidebysig specialname 
        instance void set_Item (
            valuetype [System.Runtime]System.Span`1<int64> a3,
            valuetype [System.Runtime]System.Span`1<int64> 'value'
        ) cil managed 
    {
        .param [1]
            .custom instance void [System.Runtime]System.Runtime.CompilerServices.ScopedRefAttribute::.ctor() = (
                01 00 00 00
            )

        // Method begins at RVA 0x2053
        // Code size 1 (0x1)
        .maxstack 8

        IL_0000: ret
    } // end of method Test::set_Item

    .method public hidebysig specialname rtspecialname 
        instance void .ctor () cil managed 
    {
        // Method begins at RVA 0x2055
        // Code size 7 (0x7)
        .maxstack 8

        IL_0000: ldarg.0
        IL_0001: call instance void [System.Runtime]System.Object::.ctor()
        IL_0006: ret
    } // end of method Test::.ctor

    // Properties
    .property instance valuetype [System.Runtime]System.Span`1<int64> Item(
        valuetype [System.Runtime]System.Span`1<int64> a3
    )
    {
        .get instance valuetype [System.Runtime]System.Span`1<int64> Test::get_Item(valuetype [System.Runtime]System.Span`1<int64>)
        .set instance void Test::set_Item(valuetype [System.Runtime]System.Span`1<int64>, valuetype [System.Runtime]System.Span`1<int64>)
    }

} // end of class Test
";

            var comp1 = CreateCompilationWithIL(consumer, il, targetFramework: TargetFramework.Net80);
            var indexer1 = comp1.GetMember<PropertySymbol>("Test.this[]");

            Assert.Equal(ScopedKind.ScopedValue, indexer1.Parameters[0].EffectiveScope);
            Assert.Equal(ScopedKind.ScopedValue, indexer1.SetMethod.Parameters[0].EffectiveScope);
            Assert.Equal(ScopedKind.None, indexer1.GetMethod.Parameters[0].EffectiveScope);

            comp1.VerifyDiagnostics(); // <==== EXPECT ERROR(S) HERE !
        }
```

Observed:
Errors are reported for `comp0` and no errors reported for `comp1`

Expected:
Errors for both compilation because scope of getters` parameters match in both compilations.

**Comment by AlekseyTs** [comment]
CC @cston
